### PR TITLE
Tank mod: mapgen fixes

### PR DESCRIPTION
**Pull request by chaosvolt**
Naturally I eventually discovered that quests and other content may bug out due to use of mapgen edits.
1. Removal of overmap special override, to unfuck quests and other things in the source. Instead...
2. Use of high-weight mapgen entries to make the modded version massively more likely to show up.
3. Overmap terrain and mapgen entries are retained to prevent load errors, they should be effectively dummied out.
4. Bandit camp's override remains dummied out. In its place, a much less extensive (and thus less finicky) weight-overridge for one piece of the camp has been added, to spawn the relevant bandit vehicle.
5. Adjusted weights of bandit vehiclegroup.

### CODE REVIEW

**Comment by kevingranade** [comment]
What is the problem this is trying to solve?  It would be much nicer to put the content elsewhere rather than stub it out like this.

**Comment by kevingranade** [comment]
I see, this: http://smf.cataclysmdda.com/index.php?topic=11166
Can't we just put the new bandit camp elsewhere?

**Comment by chaosvolt** [comment]
Ack. Sorry about that. I should've added the forum link.

And I suppose I could just re-implement modded bandit camps as a separate overmap special.

**Comment by chaosvolt** [comment]
While I'm at it...there's bandit cabins, bandit camps...are any of the other mapgens added relavent to quests? Those would be...mall loading bays, home improvement superstore backrooms, ranch garages, and small warehouses.

**Comment by chaosvolt** [comment]
Pbbbt. Trying to calculate the odds of a vanilla mapgen file showing up results in calculator declaring its hatred of display 1/1001 in a sane manner.

**Comment by chaosvolt** [comment]
Anyway, tested these mapgen tweaks, seem to work as intended. Also tested genning a world with the existing tank mod, saving the game in one of the soon-to-be-dummied-out areas, updating the mod folder to this version, then doing a load test. Also worked as intended.

Unless there's anything else I should tweak, I'm done mucking about with it I think.

**Comment by chaosvolt** [comment]
Ack. Weird, I could've sworn I saw that last one had passed. Oh well, farm mapgen hax aren't really essential, fixing the quest bugs was more important. Thank you for the merge.

**Comment by BevapDin** [comment]
> EDIT: Huh. Corruption or something? Never had that happen before. Any idea what went wrong for that last commit?

The build bot lost the connection with the github server and was therefor not able to build your PR. That happens sometimes, it has *nothing* to do with the PR itself (the content of the PR doesn't even make it to the server of the build bot). The build bot will (when possible) retry the PRs as soon as the master branch is changed.

**Comment by narc0tiq** [comment]
Or, if it's already been merged, won't bother.

**Comment by chaosvolt** [comment]
Ah, I see. Had me worried, but then I saw it was merged correctly, so I'd went from worried to confused. XP

File: data/mods/Tanks/dummied_out.json [hunk 1]
```diff
@@ -0,0 +1,1010 @@
+[
```
Comment (kevingranade) [inline data/mods/Tanks/dummied_out.json]: Is this spawned anywhere? If not, we are not going to drag around thousands of lines of json that doesn't actually do anything.  If it is, it needs to be renamed appropriately.

**Comment by chaosvolt** [comment]
Ack. Shouldn't spawn, no. Only exists to avoid breaking saves that genned the old overmaps before being ported over.

Though since this involved breaking quests to begin with, probably not worth the extra kilobytes to preserve the old overmaps. >w>

**Comment by chaosvolt** [comment]
In that case, shall I go ahead and PR a full removal of the dummied-out overmaps?

**Comment by chaosvolt** [comment]
Would like to know whether or not to PR that. Or if you're doing so yourself?

**Pull request merged**